components/scripts/lib/build-scan-online.sh

```diff
@@ -101,5 +101,5 @@ fetch_build_scan_data() {
     args+=( "${run_num},${build_scan_urls[run_num]}" )
   done
 
-  invoke_java "${FETCH_BUILD_SCAN_DATA_JAR}" "${args[@]}"
+  JAVA_HOME="${CLIENT_JAVA_HOME:-$JAVA_HOME}" invoke_java "${FETCH_BUILD_SCAN_DATA_JAR}" "${args[@]}"
```

Comment:
I decided to set this here instead of in `invoke_java` because invoke_java is intended to be a copy-paste of the Gradle Application Plugin's launch script (and so modifying invoke_java will risk that we accidentally overwrite this).